packages/utilities/psammead-rich-text-transforms/test/candy-xml.test.js

```diff
@@ -418,9 +413,9 @@ test('returns plain text if wrapped in an unsupport xml node', () => {
 
 test('can handle chaos', () => {
   const richText = candyXmlToRichText(
-    createBody(`
-      <paragraph><foo><bar><bold><meh>Bold text in unsupported nodes</meh></bold></bar></foo> followed by normal text then <italic><this><is>some <bold>carnage<carnage> </carnage></bold></is></this></italic>.</paragraph>
```

Comment:
How come you removed the space in carnage? Was it because you had to?

Reply:
I realised the previous logic didn't work with an empty XML tag `<carnage></carange>` so removing this space actually makes this test even more complicated

Reply:
I'll add another explicit test for an empty XML tag

Reply:
This pointed out a problem ... will push shortly

Reply:
The fragment carnage gets lost in the current implementation...
![image](https://user-images.githubusercontent.com/7791726/66573221-3b921180-eb6a-11e9-89f1-6ded14bdbcc8.png)

😞

Reply:
I don't think this should hold up this PR. I've created https://github.com/bbc/psammead/issues/2378 to address this, but I'm not convinced we will ever receive input this complicated so might not be required.